src/main/java/org/broadinstitute/hellbender/tools/copynumber/CreateBinningIntervals.java

```diff
@@ -0,0 +1,122 @@
+            minValue = 1
+    )
+    private int widthOfBins = 1;
+
+    @Argument(
+            doc = "width of the padding regions",
+            fullName = PADDING_LONG_NAME,
+            shortName = PADDING_SHORT_NAME,
+            optional = true,
+            minValue = 0
+    )
+    private int padding = 0;
+
+    @Argument(
+            doc = "output file",
+            fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+            shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+            optional = false
+    )
+    public File outputFile;
+
+
+
+    @Override
+    public void onTraversalStart() {
+        createBins();
+    }
+
+    /**
+     * Generates binning coverage in the intervals given by the user.
+     * The width of bins, the intervals and the output file's path are given by the user.
+     */
+    public void createBins() {
+        // check if the output directory exists
+        if (!outputFile.exists() && !outputFile.mkdir()) {
+            throw new RuntimeException("Unable to create file: " + outputFile.getAbsolutePath());
+        }
+
+        // check if the bin widths are set appropriately
+        if(widthOfBins <= 0) {
+            throw new IllegalArgumentException("Width of bins " + Integer.toString(widthOfBins) + " should be >= 0.");
+        }
+
+        // get the sequence dictionary
+        final SAMSequenceDictionary sequenceDictionary = getBestAvailableSequenceDictionary();
+        final List<SimpleInterval> intervals = hasIntervals() ? intervalArgumentCollection.getIntervals(sequenceDictionary)
+               : IntervalUtils.getAllIntervalsForReference(sequenceDictionary);
+
+        // create an IntervalList by copying all elements of 'intervals' into it
+        IntervalList intervalList = new IntervalList(sequenceDictionary);
+        intervals.stream().map(si -> new Interval(si.getContig(), si.getStart(), si.getEnd())).forEach(intervalList::add);
+
+        // sort intervals according to their coordinates and unique them (i.e. delete duplicates)
+        intervalList.uniqued();
+
+        // pad all elements of intervalList
+        intervalList = intervalList.padded(padding,padding);
+
+        // merge those that intersect after padding
+        intervalList = IntervalList.intersection(intervalList, intervalList);
+
+        // break the intervals up to bins -- the last bin in each interval can be shorter than the others
+        IntervalList bins = new IntervalList(sequenceDictionary);
+        int bin_start, bin_end;
+        Interval new_bin;
+        for(Interval in : intervalList) {
+            bin_start = in.getStart();
+            bin_end = Math.min(bin_start + widthOfBins - 1, in.getEnd());
+            while(bin_start < in.getEnd()) {
+                new_bin = new Interval(in.getContig(), bin_start, bin_end);
+                bins.add(new_bin);
+                bin_start += widthOfBins;
+                bin_end = Math.min(bin_start + widthOfBins - 1, in.getEnd());
```

Comment:
If you move `bin_end`'s declaration into the loop this line becomes unnecessary, I think.

Reply:
I did that, thx.